### PR TITLE
Add logging + counter + jitter to view redirects to prevent infinite redirect loop

### PR DESCRIPTION
**Pull request by Nezteb**
Attempts to fix https://github.com/phoenixframework/phoenix_live_view/issues/3379

### CODE REVIEW

File: test/phoenix_component_test.exs [hunk 1]
```diff
@@ -332,7 +332,7 @@ defmodule Phoenix.ComponentUnitTest do
       "bday" => %{"day" => "", "month" => "", "year" => ""},
       "published_at" => %{"date" => "", "time" => "", "_unused_date" => "", "_unused_time" => ""},
       "deleted_at" => %{},
-      "inserted_at" => %{"date" => "", "time" => "", "_unused_time" => ""},
+      "inserted_at" => %{"date" => "", "time" => "", "_unused_time" => ""}
```
Comment (Nezteb) [inline test/phoenix_component_test.exs:335]: From `mix format`

File: assets/js/phoenix_live_view/view.js [hunk 2]
```diff
@@ -828,6 +839,27 @@ export default class View {
...
+  
+    this.redirectWithJitterTimer = setTimeout(() => {
+      this.log("error", () => [`encountered ${redirectAttempts} consecutive redirects`])
+      if (redirectAttempts > this.maxRedirects) {
+        this.log("error", () => [`exceeded ${this.maxRedirects} consecutive redirects. Entering failsafe mode`])
+      }
+      if (this.isMain()) {
+        this.onRedirect({ to: this.href })
+      }
+    }, afterMs)
```
Comment (Nezteb) [inline assets/js/phoenix_live_view/view.js:860]: For some reason this timeout function isn't running.

I copied the general pattern from https://github.com/phoenixframework/phoenix_live_view/blob/main/assets/js/phoenix_live_view/live_socket.js#L312-L336

File: assets/js/phoenix_live_view/constants.js [hunk 3]
```diff
@@ -2,7 +2,12 @@ export const CONSECUTIVE_RELOADS = "consecutive-reloads"
 export const MAX_RELOADS = 10
 export const RELOAD_JITTER_MIN = 5000
 export const RELOAD_JITTER_MAX = 10000
-export const FAILSAFE_JITTER = 30000
+export const RELOAD_FAILSAFE_JITTER = 30000
+export const CONSECUTIVE_REDIRECTS = "consecutive-redirects"
+export const MAX_REDIRECTS = 3
+export const REDIRECT_JITTER_MIN = 1000
+export const REDIRECT_JITTER_MAX = 5000
+export const REDIRECT_FAILSAFE_JITTER = 20000
```
Comment (Nezteb) [inline assets/js/phoenix_live_view/constants.js:10]: I could have re-used the `RELOAD_` constants but opted to make new ones specifically for redirects.

**Comment by SteffenDE** [comment]
Hi there! We also have https://github.com/phoenixframework/phoenix_live_view/pull/3280 open at the moment, which also touches the error handling. So I'd probably wait with this one for a little bit. Maybe we'll also try to incorporate a fix for #3379 in there.

And btw, please don't include the compiled assets in priv/static in your PR, we'll generate them afterwards :)

**Comment by Nezteb** [comment]
Sounds good! I reverted the static asset updates.

I'll watch your PR in the meantime!

**Comment by Nezteb** [comment]
Closing in favor of https://github.com/phoenixframework/phoenix_live_view/pull/3459 😄

**Comment by chrismccord** [comment]
Thanks for looking into this! I really appreciate anyone who deep dives into this stuff so much appreciated ❤️

**Pull request closed without merging**